### PR TITLE
JobStore read-side skeleton

### PR DESCRIPTION
**Pull request by cjllanwarne**
This is obviously a long way from being feature-complete, but is hopefully enough of a skeleton/statement of intent for JobStore readers to develop against in the meanwhile...

### CODE REVIEW

**Comment by cjllanwarne** [comment]
@Horneth should be relevant for your EJEA stuff. Needs one other reviewer (whoever fancies it, or otherwise I'll wheel for it)

File: engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  override def receive = LoggingReceive {
+    case IsJobCompletedYet(jobKey) => sender ! JobNotCompleteYet(jobKey)
+    case unknownMessage => log.error(s"Unexpected message to JobStoreReader: $unknownMessage")
+  }
+}
+
+case class IsJobCompletedYet(jobKey: JobKey)
+
+case class JobAlreadyCompleted(jobKey: JobKey, jobResult: JobResult)
+case class JobNotCompleteYet(jobKey: JobKey)
```
Comment (Horneth) [inline engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala]: Wouldn't `JobComplete` `JobRunning` be descriptive enough ?

**Comment by Horneth** [comment]
LGTM, if we realize something is missing implementing EJEA I'll drop a ticket 👍

File: engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala (hunk 1 above)
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala]: I guess I could change it to `JobComplete` and `JobNotComplete`? I think there's no reason to distinguish running vs unstarted here, that would be a backend issue at restore-time

Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala]: Yeah I like `JobComplete` and `JobNotComplete`.  `JobRunning` seems to imply a more specific knowledge of state that I don't think this code will know?

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**